<launch>
    <arg name="repeat_rate" default="40.0" />
    <arg name="joy_config" default="$(find arka_manipulator)/config/joy_new_config.yaml"/>
    <arg name="arka_arm_config" default="$(find arka_manipulator)/config/manipulator_joy.yaml"/>

    <node pkg="arka_manipulator" name="manipulator" type="manipulator.py" output="screen" respawn="true" respawn_delay="1">
        <!-- <rosparam command="load" file="$(arg arka_config)" /> -->
        <param name="controller_ip"   type="string" value="192.168.10.10"/>
        <param name="is_manip_enable" type="bool" value="true" />
        <param name="is_semi_enable" type="bool" value="false" />
        <param name="main_board_ip" type="string" value="192.168.10.170" />
        <param name="main_board_port" type="string" value="3030" />
        <param name="sensor_board_ip" type="string" value="192.168.10.20" />
        <param name="sensor_board_port" type="string" value="3030" />
        <param name="joint1/ratio" type="int" value="2825" />
        <param name="joint2/ratio" type="int" value="3900" />
        <param name="joint3/ratio" type="int" value="2825" />
        <param name="joint4/ratio" type="int" value="1" />
        <!-- <param name="arm_calibration_path" type= "string" value="$(find arka_manipulator)/config/manual_calibration.yaml" /> -->
    </node>

    <node pkg="joy" type="joy_node" name="ros_joy" respawn="true" respawn_delay="1">
        <param name="autorepeat_rate" value="$(arg repeat_rate)" />
    </node>

    <node pkg="mercury_joy" type="joy-node.py" name="mercury_joy" respawn="true" respawn_delay="1">
        <param name="mercury_publish_rate" value="$(arg repeat_rate)" />
    </node>

    <node name="dynamic_reconfigure_load_joy" pkg="dynamic_reconfigure" type="dynparam" args="load /mercury_joy $(arg joy_config)" />
    <node name="dynamic_reconfigure_load_arm_cfg" pkg="dynamic_reconfigure" type="dynparam" args="load /manipulator $(arg arka_arm_config)" />

</launch>
